crmsh/bootstrap.py

```diff
@@ -491,6 +532,46 @@ def get_sbd_device_from_config(cls):
         inst = cls()
         return inst._get_sbd_device_from_config()
 
+    @classmethod
+    def is_using_diskless_sbd(cls):
+        """
+        Check if using diskless SBD
+        """
+        inst = cls()
+        dev_list = inst._get_sbd_device_from_config()
+        if not dev_list and utils.service_is_active("sbd.service"):
+            return True
+        return False
+
+    @staticmethod
+    def update_configuration(sbd_config_dict):
+        """
+        Update and sync sbd configuration
+        """
+        utils.sysconfig_set(SYSCONFIG_SBD, **sbd_config_dict)
+        csync2_update(SYSCONFIG_SBD)
+
+    @staticmethod
+    def calculate_stonith_timeout(sbd_watchdog_timeout, check_exist=False):
+        """
+        Calculate stonith timeout
+        """
+        stonith_timeout = int(sbd_watchdog_timeout * 2 * 1.2)
+        if check_exist:
+            exist_st = cib_factory.get_property("stonith-timeout")
+            if exist_st and int(exist_st.strip('s')) >= stonith_timeout:
+                stonith_timeout = int(exist_st.strip('s'))
```

Comment:
Similarly, users could configure this with their favorite units too. It's kind of difficult to parse the value for comparison. Let's make it easy to just use `sbd_watchdog_timeout * 2 * 1.2`.

Reply:
OK
Changed